migrations/extract-data-record-content.js

```diff
@@ -0,0 +1,102 @@
+var _ = require('underscore'),
+    db = require('../db'),
+    async = require('async');
+
+var BATCH_SIZE = 100;
+var PERCENT_REPORT_CHUNKS = 10;
+
+// Copied from https://github.com/node-browser-compat/btoa/blob/master/index.js
+// because why import a library for one tiny function
+function btoa(str) {
+  var buffer;
+
+  if (str instanceof Buffer) {
+    buffer = str;
+  } else {
+    buffer = new Buffer(str.toString(), 'binary');
+  }
+
+  return buffer.toString('base64');
+}
+
+var attachmentIse = function(record) {
+  // XML forms which have not been migrated yet
+  if (record.content_type === 'xml' && record.content) {
+    record._attachments = record._attachments || {};
+    record._attachments.content = {
+      content_type: 'application/xml',
+      data: btoa(record.content)
+    };
+    delete record.content;
+
+    return record;
+  }
+};
+
+
+var incrementPercentTarget = function(percentTarget) {
+  return percentTarget + PERCENT_REPORT_CHUNKS;
+};
+
+module.exports = {
+  name: 'extract-data-record-content',
+  created: new Date(2017, 0, 6),
+  run: function(callback) {
+    console.log('Finding data_records...');
+    db.medic.view('medic', 'data_records', function(err, body) {
```

Comment:
I think fetching _all_ data record stubs will cause issues on large DBs (eg: LG). Because of this, other migrations use limit and skip to batch the view request and include docs at the same time. You get `total_rows` in the view response so you can still work out percentage complete (which is nice btw!). You'd have to change your logic regarding done a little, but overall it'd be more reliable.